### PR TITLE
Various improvements + Updated tests

### PR DESCRIPTION
**Pull request by vladrusu**
- Better emoticon handling
- Better clickable & email handling. For simplified logic, replace is done now before parse.
- Ability to support both attributes surrounded by quota or not in non-strict mode for better integration with various BB code editors (like SCEditor)
- Added "decoda-emoticon" class to emoticons in order to be able to style them in code
- Added "decoda-image" class to &lt;img&gt;s in order to be able to style them in code

### CODE REVIEW

**Comment by vladrusu** [comment]
Hello !
I put all the improvements I made in one big PR.
Hope it suits you! ;-)
Regards,
Vlad

File: tests/Decoda/Hook/EmoticonHookTest.php [hunk 1]
```diff
@@ -28,13 +28,6 @@ protected function setUp() {
         $this->object->addFilter(new DefaultFilter());
 
         $hook = new EmoticonHook();
-        $hook->addLoader(new DataLoader(array(
```
Comment (milesj) [inline tests/Decoda/Hook/EmoticonHookTest.php]: Reason for this removal?

**Comment by milesj** [comment]
Great work so far, thanks for the improvements!

Just one minor question.

**Comment by vladrusu** [comment]
Hello !
Thanks!
I added back the custom emoticons tests (sorry, my mistake), with minor adjustments.
Also, I added a test for the unicode emoticon ☺ ... which initially missed from your test.

**Comment by milesj** [comment]
Sweet, thanks a lot for the improvements!
